### ISSUE TITLE
Kodi crashes if a youtube video is shared through Kore while the previous video is still in the throbber stage

### ISSUE THREAD

**Issue by Managor** [xbmc/xbmc#20468]
### Describe the bug
Here is a clear and concise description of what the problem is:

When you share a youtube video to kodi to be played by Plugin for Youtube, sometimes the throbber loading screen takes a while longer. If you share another video to be played or queued while this is happening, whole kodi will crash

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

No crash should happen

## Actual Behavior

Kodi should abandon the previous video and play the new one or put the new video at the back of the queue respectively

### To Reproduce

1. Find a youtube video that takes a longer time to buffer on Kodi
2. Select to share a video to Kodi through Kore
3. Take any other video and share it to Kodi

### Debuglog

The debuglog can be found here:
[kodi_crashlog-20211007_141534.log](https://github.com/xbmc/xbmc/files/7482113/kodi_crashlog-20211007_141534.log)

### Screenshots 
Here are some links or screenshots to help explain the problem:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Arch Linux
 - Kodi version: 19.3.0

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by ccoenen** [comment]
can confirm, this happens to me all the time :-(

**Comment by ccoenen** [comment]
this is a bug regardless of activity.

**Comment by garbear** [comment]
Busy dialog crash should be fixed by https://github.com/xbmc/xbmc/pull/27038, right? Anyone can try the test builds linked from the PR?

**Comment by ccoenen** [comment]
this bug still happens to me once every couple of days.

**Comment by garbear** [comment]
@ccoenen Are you able to try a test build? https://github.com/garbear/xbmc/releases/tag/retroplayer-22alpha1-20250803 - these should fix the problem, and if confirmed we can merge the fix into master.

**Comment by ccoenen** [comment]
Your link looks like a full libreElec img, is that correct? I am not sure if I can test this right away. I don't want to ruin my (otherwise working) regular libreElec install :-D

If I'm missing something obvious, please point it out. I might totally have misunderstood something! I haven't contributed to this particular project beyond very simple comments.

**Comment by Nanomani** [comment]
> Your link looks like a full libreElec img, is that correct? I am not sure if I can test this right away. I don't want to ruin my (otherwise working) regular libreElec install :-D
> 
> If I'm missing something obvious, please point it out. I might totally have misunderstood something! I haven't contributed to this particular project beyond very simple comments.

All OS here : 
https://github.com/garbear/xbmc/releases

**Comment by ccoenen** [comment]
I have tried to set everything up, but I didn't manage to even test the bug. For some reason I can't log in to YouTube (yes I have created API keys). I will try again at the next opportunity. Sorry that this is taking so long.

**Comment by ccoenen** [comment]
Yeah, for some reason I can't seem to activate my Youtube API key in a way that lets me actually log in. I think that particular fault is on my end, not yours. But it also means: I will not be able to test this. Sorry!

**Comment by neo1973** [comment]
Closed because probably fixed by #27038.
